### PR TITLE
Added VQFN-48-1EP_7x7mm_P0.5mm_EP5.15x5.15mm

### PR DESCRIPTION
**Pull request by ppielorz**
Added new entry in script vqfn.yaml according to TI CC1312R datasheet.

### CODE REVIEW

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 1]
```diff
@@ -505,3 +505,68 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
   #pad_length_addition: 0.5
   #suffix: '_Pad{pad_x:.2f}x{pad_y:.2f}mm_HandSolder'
   #include_suffix_in_3dpath: 'False'
+
+
+VQFN-48-1EP_7x7mm_P0.5mm_EP5.15x5.15mm:
+  device_type: 'VQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'http://www.ti.com/lit/ds/symlink/cc1312r.pdf'
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: please add #page=48 to point directly to the correct page.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 2]
```diff
@@ -505,3 +505,68 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
...
+  ipc_class: 'qfn' # 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
+  # custom_name_format:
+  body_size_x:
+    minimum: 6.9
+    nominal: 7
+    maximum: 7.1
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: nominal size is not given in the datasheet so please remove here.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 3]
```diff
@@ -505,3 +505,68 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
...
+  overall_height:
+    #minimum: 0.8
+    #nominal: 0.85
+    maximum: 1
+  lead_width:
+    #minimum: 0.18
+    nominal: 0.24
+    #maximum: 0.3
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: given as 0.18 ... 0.3 so please use this here

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 4]
```diff
@@ -505,3 +505,68 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
...
+  lead_len:
+    #minimum: 0.25
+    nominal: 0.3
+    #maximum: 0.35
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: given as 0.3 ... 0.5 so use this here

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 5]
```diff
@@ -505,3 +505,68 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
...
+
+  EP_size_x:
+    minimum: 5.05
+    nominal: 5.15
+    maximum: 5.25
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: given as 5.15 with +/- tolerance so use the nominal plus tolerance style.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 6]
```diff
@@ -505,3 +505,68 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
...
+  EP_size_y:
+    minimum: 5.05
+    nominal: 5.15
+    maximum: 5.25
+  # EP_paste_coverage: 0.65
+  EP_num_paste_pads: [2, 2]
+  # heel_reduction: 0.05 #for relatively large EP pads (increase clearance)
+
+  thermal_vias:
+    count: [3, 3]
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: datasheet would suggest [5, 5]

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml [hunk 7]
```diff
@@ -505,3 +505,68 @@ VQFN-32-1EP_5x5mm_P0.5mm_EP3.1x3.1mm:
...
+    drill: 0.33
+    # min_annular_ring: 0.15
+    paste_via_clearance: 0.1
+    # EP_num_paste_pads: [2, 2]
+    # paste_between_vias: 1
+    # paste_rings_outside: 1
+    EP_paste_coverage: 0.7
+    grid: [1.2, 1.2]
```
Comment (poeschlr) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/vqfn.yaml]: datasheet suggests a very strange layout not supported by the script so i would suggest to drop this and let the script decide automatically how to place vias.

**Comment by ppielorz** [comment]
It should be fine now - forgot about y axis and one small mistake.

**Comment by poeschlr** [comment]
The script changes look good now. You however forgot to update the footprint pull request.

**Comment by poeschlr** [comment]
thanks

**Pull request merged**